db/dbScript.sql

```diff
@@ -0,0 +1,393 @@
+  website VARCHAR(255),
+  is_wheelchair_accessible BOOL NOT NULL,
+  membership_price NUMERIC(5,2) check (membership_price >= 0),
+  created_at TIMESTAMP NOT null check (created_at <= NOW()),
+  price_changed_at TIMESTAMP check (price_changed_at >= changed_at and price_changed_at > created_at),
+  changed_at TIMESTAMP check (changed_at > created_at and changed_at <= NOW()),
+  internal_rating NUMERIC(4,2) NOT null check (internal_rating between 0 and 5), --local rating by local users
+  internal_rating_number INT NOT null check (internal_rating_number >= 0), --number of accounts that contributed to the local rating
+  congestion_rating NUMERIC(4,2) NOT null check (congestion_rating between 0 and 5),
+  congestion_rating_number INT NOT null check (congestion_rating_number >= 0),
+  owned_by uuid, --a gym account that manages information about this gym
+  currency_id uuid NOT NULL,
+  city_id uuid not null,
+  PRIMARY KEY (id),
+  FOREIGN KEY (owned_by) REFERENCES Account(id),
+  FOREIGN KEY (currency_id) REFERENCES Currency(id),
+  foreign key (city_id) references City(id),
+  UNIQUE (external_place_id)
+);
+create index if not exists idx_gym_external_place_id on Gym(external_place_id);
+create index if not exists idx_gym_lat_lon on Gym(latitude, longitude);
+create index if not exists idx_gym_price_changed_at on Gym(price_changed_at);
+create index if not exists idx_gym_owned_by on Gym(owned_by);
+create index if not exists idx_gym_city_id on Gym(city_id);
+
+CREATE TABLE if not exists Ownership
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),
+  requested_at TIMESTAMP NOT null check (requested_at <= NOW()), --time at which a gym account requested the ownership
+  responded_at TIMESTAMP check (responded_at > requested_at and responded_at <= NOW()), --time at which an admin made a decision regarding the ownership
+  status own_status NOT NULL, --status of the requested ownership: pending, approved, rejected
+  message text, --message the admin might write in order to explain his/her decision
+  responded_by uuid, --the admin account that made the decision
+  requested_by uuid NOT NULL, --the gym account that made the ownership request
+  gym_id uuid  NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (responded_by) REFERENCES Account(id),
+  FOREIGN KEY (requested_by) REFERENCES Account(id),
+  FOREIGN KEY (gym_id) REFERENCES Gym(id),
+  UNIQUE (gym_id, requested_by, status)
+);
+
+CREATE TABLE if not exists Request
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),
+  requested_at TIMESTAMP NOT null check (requested_at <= NOW()), --time of the request
+  origin_latitude FLOAT NOT null check (origin_latitude between -90 and 90), --latitude of the starting location
+  origin_longitude FLOAT NOT null check (origin_longitude between -180 and 180),
+  departure_time TIME, --preferred departure_time specified by user
+  weekday INT check (weekday between 0 and 6), --day of the week for which the user wants to receive the recommendations 
```

Comment:
Yeah, I do think it makes sense. In the long run, it will allow us to probably minimize the number of requests we need to make and make the recommendation even more tailored to the customer. Plus we can make these requests (we will have to make a separate request for each day of the week regarding the selected time) in parallel